ebi_eva_common_pyutils/assembly/assembly.py

```diff
@@ -29,3 +31,39 @@ def get_supported_asm_from_ensembl(tax_id: int) -> str:
     if assembly_accession_attribute in response:
         return str(response.get(assembly_accession_attribute))
     return None
+
+
+@lru_cache
+def get_taxonomy_to_assembly_mapping_from_ensembl_rapid_release():
+    """
+    Returns a dict mapping taxonomy ID to assembly accession, choosing the most recently released,
+    lexicographically first, non-alternate haplotype assembly when multiple are present.
+    """
+    list_data = json_request('https://ftp.ensembl.org/pub/rapid-release/species_metadata.json')
+    results = {}
+    for asm_data in list_data:
+        tax_id = asm_data['taxonomy_id']
+        asm_accession = asm_data['assembly_accession']
+        strain = asm_data['strain']
+        release_date = datetime.strptime(asm_data['release_date'], '%Y-%m-%d')
+
+        # If we haven't seen this taxonomy before, just use this assembly
+        if tax_id not in results:
+            results[tax_id] = (asm_accession, release_date)
+            continue
+
+        # Skip alternate haplotype assemblies
+        if strain and strain.lower() == 'alternate haplotype':
+            continue
+        current_assembly, current_date = results[tax_id]
+        # Keep the more recent assembly, or the lexicographically first one if release dates are equal
+        if current_date < release_date or (current_date == release_date and asm_accession < current_assembly):
```

Comment:
**Optional:** Guess you have already run an experiment to see if there are cases where release date for an older assembly is more recent than the release date for a newer assembly for some reason (an update/correction to the older assembly perhaps?)

Reply:
I've not actually checked this, do you mean comparing against dates in NCBI for example? I'm not sure it's worth doing since this is intended as a short-term measure till Ensembl makes the API public, but we could perhaps ask Ensembl if the scenario you proposed is plausible.

Reply:
cc @tcezard As discussed, this does occasionally occur. Here's an example where the later patch has an earlier release date:

![Screenshot from 2023-09-21 14-53-42](https://github.com/EBIvariation/eva-common-pyutils/assets/5599307/1c44de63-5a25-4539-8eaa-2f5fcf9cf266)

And an example where two patch versions have the same release date:

![Screenshot from 2023-09-22 10-17-02](https://github.com/EBIvariation/eva-common-pyutils/assets/5599307/2b762f22-8cbc-4f0f-9600-e81ef41d36e7)

I modified the lexicographical ordering to choose the last rather than the first to account for the second case.

Sundar suggested we could also look at the `geneset` attribute, but I think at this point it would be preferable to ask Ensembl for clarification and make any necessary changes in a subsequent PR.

Reply:
makes sense...